packages/core/src/storage/StorageUtility.ts

```diff
@@ -116,9 +114,6 @@ export async function saveSessionInfoToStorage(
   if (refreshToken !== undefined) {
     await storageUtility.setForUser(sessionId, { refreshToken }, { secure });
   }
-  if (idToken !== undefined) {
-    await storageUtility.setForUser(sessionId, { idToken }, { secure });
```

Comment:
I think we may also need to explicitly clear the key for the idToken here too, such that we don't accidentally leave dead data in localstorage

Reply:
Agree on clearing local storage, but I think it could happen in a different place. Session data is currently cleared when the user logs out, but that means that it hangs around if the user just leaves without properly logging out. Note that the session ID is a randomly generated uuid, which means that logging in without logging out multiple times leaves multiple instances of session data hanging around, which isn't great. It's not really an issue on the server side with the Node codebase, because we want users to be able to be logged back in silently, but that's not possible on the client-side because it's unsafe to persist the refresh token in local storage.

All this to say, what we could do instead is, for `solid-client-authn-browser`, on login, go through local storage and clear all session data associated to the same webid. This way, we're sure that the local storage doesn't grow infinitely, and that any stale data is removed from there. If you don't mind, I'd rather do this in a separate PR, because it's pretty much its own thing. Does that sound good to you ?

Reply:
I think I need more time to better understand this, but I think we could also store a `session:${id}:expiresAfter` date/time, and then flush (clear) all sessions where expiresAfter is `< Date.now()`

I would also like to see this behaviour very clearly documented, if it is not already, as this aspect of the SDK is certainly where we can get really bad privacy bugs forming. e.g., when you sign-in, should there be a "remember me" checkbox, and if it's not checked then we don't store stuff past the current browsing session — single sessionId mode)

Reply:
+1 on the documenting this, I was just thinking about adding a section in the `ARCHITECTURE.md` about exactly what gets stored where.

Reply:
And also, expiring the session is a bit challenging: what I would be afraid of is to end up in a situation where a session which is still active gets logged out. The session being "active" means it has a valid access token basically, which happens internally in the fetch closure. What could be done is to inject a callback all the way into the fetch so that they have a reference to the storage and keep the expiry date current. I'll try to design this properly so that we can talk about it :)